### ISSUE TITLE
Cache for Light Change Timings to Music

### ISSUE THREAD

**Issue by hammy275** [hammy275/rgb-light-control#5]
The cache would be a folder called `cache`. Inside of this would be a folder called `music`, which finally contains the folders `cycle` and `gradient` for the two music modes. Both of those folders contain a file named `version.json`, which contains the version number used, so that when the function is rewritten, the caches can be automatically invalidated.

Each folder also will consist of JSON files named `HASH1-HASH2.json`. `HASH1` is replaced by the `filepath`, while `HASH2` is replaced by the `calc_filepath`. 

Files in the `cycle` folder have the following structure
```json
{
    "timings": "An array of times within the song to do a light change"
}
```

Files in the `gradient` folder have the following structure
```json
{
    "timings": "An array of times within the song to do a light change",
    "weights": "An array of weights for the timing at the same index."
}
```

When loading a song, it will first hash both files. If a cache file is found with matching hashes, it will load data from it instead of re-calculating.

On program startup, all cache directories are made, and all `version.json` files are checked if they exist. If one does exist but is showing an older version, the cache folder for that out-of-date mode is wiped clean.

Some notes on this:
- The gradient method will have to be reworked, since it currently doesn't progress weights through the function.
- The gradient method is based on the average light change time. This should either be changed, or the program should be reworked to account for bad ping conditions by not `await`ing light sends.
- There should be a mode to generate caches for all songs in a library once #2 is done, but that's out of scope of this issue.
